s3client.go

```diff
@@ -21,7 +21,7 @@ type S3Client interface {
 	BucketFiles(bucketName string, prefixHint string) ([]string, error)
 	BucketFileVersions(bucketName string, remotePath string) ([]string, error)
 
-	UploadFile(bucketName string, remotePath string, localPath string, acl string, serverSideEncryption string, kmsKeyId string) (string, error)
+	UploadFile(bucketName string, remotePath string, localPath string, acl string, contentType string, serverSideEncryption string, kmsKeyId string) (string, error)
```

Comment:
I think it's time to refactor this; the argument list keeps growing and growing. Maybe we could pass in an `UploadInput` with fields set and just have `UploadFile` fill in the `Body` attribute?